lib/chromedriver.js

```diff
@@ -512,72 +521,45 @@ export class Chromedriver extends events.EventEmitter {
   }
 
   /**
-   * Sync the WebDriver protocol if current on going protocol is W3C or MJSONWP.
-   * Does nothing if this.driverVersion is null.
+   * Determines the driver communication protocol
+   * based on various validation rules.
    *
-   * @returns {typeof PROTOCOLS[keyof typeof PROTOCOLS]}
+   * @returns {keyof PROTOCOLS}
    */
   syncProtocol() {
-    if (!this.driverVersion) {
-      // Keep the default protocol if the driverVersion was unsure.
-      return this.desiredProtocol;
+    if (this.driverVersion) {
+      const coercedVersion = semver.coerce(this.driverVersion);
+      if (!coercedVersion || coercedVersion.major < MIN_CD_VERSION_WITH_W3C_SUPPORT) {
+        this.log.info(
+          `The ChromeDriver v. ${this.driverVersion} does not fully support ${PROTOCOLS.W3C} protocol. ` +
+            `Defaulting to ${PROTOCOLS.MJSONWP}`,
+        );
+        this._desiredProtocol = PROTOCOLS.MJSONWP;
+        return this._desiredProtocol;
+      }
     }
 
-    this.desiredProtocol = PROTOCOLS.MJSONWP;
-    const coercedVersion = semver.coerce(this.driverVersion);
-    if (!coercedVersion || coercedVersion.major < MIN_CD_VERSION_WITH_W3C_SUPPORT) {
-      this.log.info(
-        `The ChromeDriver v. ${this.driverVersion} does not fully support ${PROTOCOLS.W3C} protocol. ` +
-          `Defaulting to ${PROTOCOLS.MJSONWP}`,
-      );
-      return this.desiredProtocol;
-    }
-    // Check only chromeOptions for now.
-    const chromeOptions = getCapValue(this.capabilities, 'chromeOptions', {});
-    if (chromeOptions.w3c === false) {
+    const isOperaDriver = _.includes(this._onlineStatus?.message, 'OperaDriver');
+    const chromeOptions = getCapValue(this.capabilities, 'chromeOptions');
```

Comment:
Do you think it would be nice to add `ms:edgeOptions` for MSEdge as well?
MSEdge supports `chormeOptions` naming still, so I don't have any strong opinion for this. I don't remember well, but probably edge prior edgeOptions than chormeOptions

Reply:
edgeOptions map does not contain the `w3c` flag, so I assume this logic could remain unchanged for now: https://learn.microsoft.com/de-de/microsoft-edge/webdriver-chromium/capabilities-edge-options